docs/data-sources/privileges.md

```diff
@@ -0,0 +1,70 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "googleworkspace_privileges Data Source - terraform-provider-googleworkspace"
+subcategory: ""
+description: |-
+  Privileges data source in the Terraform Googleworkspace provider.
+---
+
+# googleworkspace_privileges (Data Source)
+
+Privileges data source in the Terraform Googleworkspace provider.
+
+## Example Usage
+
+```terraform
+data "googleworkspace_privileges" "privileges" {}
+
+# filter using native Terraform syntax
+locals {
+  org_scopable_privileges = [
+    for priv in data.googleworkspace_privileges.privileges.items : priv
+    if priv.is_org_unit_scopable
+  ]
+  read_only_privileges = [
+    for priv in data.googleworkspace_privileges.privileges.items : priv
+    if length(regexall("READ", priv.privilege_name)) > 0
+  ]
+  privileges_by_service_name = [
+    for priv in data.googleworkspace_privileges.privileges.items : priv
+    if priv.service_name == "gmail"
+  ]
+}
+
+resource "googleworkspace_role" "read-only" {
+  name = "read-only"
+
+  dynamic "privileges" {
+    for_each = local.read_only_privileges
+    content {
+      service_id     = privileges.value["service_id"]
+      privilege_name = privileges.value["privilege_name"]
+    }
+  }
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Optional
+
+- **id** (String) The ID of this resource.
+
+### Read-Only
+
+- **etag** (String) ETag of the resource.
+- **items** (List of Object) A list of Privilege resources. (see [below for nested schema](#nestedatt--items))
```

Comment:
do we want to mention somewhere that this will be a flattened list?

Reply:
will do